### PR TITLE
Burnt matches no longer permanently give you a light source if they naturally burn out

### PR DESCRIPTION
**Pull request by morrowwolf**
# About the pull request

Burnt matches no longer permanently give you a light source if they naturally burn out

Reviewer note: I guess I could just like set up a variable for the user, iunno happy to do it whatever way

# Explain why it's good for the game

Bug bad

# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog

:cl: Morrow
fix: Burnt matches no longer permanently give you a light source if they naturally burn out
/:cl:

### CODE REVIEW

**Review by fira — APPROVED** [review]

File: code/game/objects/items/tools/flame_tools.dm [hunk 1]
```diff
@@ -170,7 +173,7 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 	damtype = "brute"
 	icon_state = "[initial(icon_state)]_burnt"
 	item_state = "cigoff"
-	if(user && loc != user)
+	if(user)
```
Comment (Zonespace27) [inline code/game/objects/items/tools/flame_tools.dm:176]:
```suggestion
	if(ismob(loc))
```

File: code/game/objects/items/tools/flame_tools.dm [hunk 2]
```diff
@@ -170,7 +173,7 @@ CIGARETTE PACKETS ARE IN FANCY.DM
...
 		user.SetLuminosity(0, FALSE, src)
```
Comment (Zonespace27) [inline code/game/objects/items/tools/flame_tools.dm:177]: And then above here add 'var/mob/user = loc' while removing the arg from the proc

File: code/game/objects/items/tools/flame_tools.dm [hunk 3]
```diff
@@ -136,7 +136,10 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 /obj/item/tool/match/process(delta_time)
 	smoketime -= delta_time SECONDS
 	if(smoketime < 1)
-		burn_out()
+		var/user
+		if(istype(loc, /mob))
+			user = loc
+		burn_out(user)
```
Comment (Zonespace27) [inline code/game/objects/items/tools/flame_tools.dm:142]: Undo your changes here

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

**Comment by morrowwolf** [comment]
Ran into an issue where when dropping it the light wouldn't go out. Left the argument and just added a check after the user check to see if loc is a mob. I'm not sure this is more or less elegant but 🤷

**Review by Zonespace27 — APPROVED** [review]
Sure ig
Luminosity should really depend on a trait or component so multiple sources can exist but thats outta scope

**Pull request merged**